nomos-core/src/da/certificate/verify.rs

```diff
@@ -0,0 +1,210 @@
+use std::marker::PhantomData;
+
+use crate::da::{attestation::Attestation, auth::Verifier, certificate::Certificate};
+
+use super::CertificateVerifier;
+
+pub trait KeyStore {
+    type Key;
+    type Verifier: Verifier;
+
+    fn get_key(&self, node_id: &Self::Key) -> Option<&Self::Verifier>;
```

Comment:
maybe this should be `get_verifier` or just `get`? and I think the parameter name needs to be `key` instead of `node_id` because the type `Key` is generic.

Reply:
Ah, now I understand that the `key` of the `get_key` means public key, and the type `Key` means the key of a key-value pair.